pkgs/inject_dartpad/lib/inject_dartpad.dart

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) 2025, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:js_interop';
+
+import 'package:web/web.dart' as web;
+
+/// An iframe-embedded DartPad that can be injected into a web page,
+/// then have its source code updated.
+///
+/// Example usage:
+///
+/// ```dart
+/// import 'package:inject_dartpad/inject_dartpad.dart';
+/// import 'package:web/web.dart' as web;
+///
+/// void main() async {
+///   final dartPad = EmbeddedDartPad.create(
+///     iframeId: 'my-dartpad',
+///     theme: DartPadTheme.light,
+///   );
+///
+///   await dartPad.initialize(
+///     addToDocument: (iframe) {
+///       iframe.style.height = '560';
+///
+///       web.document.body!.append(iframe);
+///     },
+///   );
+///
+///   dartPad.updateCode('''
+/// void main() {
+///   print("Hello, I'm Dash!");
+/// }''');
+/// }
+/// ```
+final class EmbeddedDartPad {
+  /// The unique identifier that's used to identify the created DartPad iframe.
+  ///
+  /// This ID is used both as the HTML element `id` and
+  /// as the iframe's `name` attribute for message targeting.
+  final String iframeId;
```

Comment:
This makes it seem like the user needs to put an iframe on the page, but my understanding is that this needs to be a `<code>` element, right? Maybe we should make this just `elementId` if that's the case.

Reply:
In the `initialize` function the user of the class will need to put a created iframe (with this ID) onto the page.

The `code` element is just required by the script in `web/inject_dartpad.dart`.